test/test_distributions.py

```diff
@@ -3423,3 +3425,65 @@ def test_gaussian_random_walk_linear_recursive_equivalence():
     x2 = dist2.sample(random.PRNGKey(7))
     assert jnp.allclose(x1, x2.squeeze())
     assert jnp.allclose(dist1.log_prob(x1), dist2.log_prob(x2))
+
+
+@pytest.mark.parametrize(
+    "my_dist",
+    [
+        dist.TruncatedNormal(low=-1, high=1),
+        dist.TruncatedCauchy(low=-1, high=1),
+    ],
+)
+def test_tracer_leakage_in_parallel_truncated_distribution(my_dist):
+    n_data_samples = 1
+    n_mcmc_samples = 10
+    n_mcmc_warmup = 10
+    n_mcmc_chains = 2
+    rng_key = jax.random.PRNGKey(0)
+
+    data_samples = my_dist.sample(rng_key, (n_data_samples,))
+
+    def my_model():
+        numpyro.sample("obs", my_dist, obs=data_samples)
+
+    nuts_kernel = NUTS(my_model)
+    mcmc = MCMC(
+        nuts_kernel,
+        num_warmup=n_mcmc_warmup,
+        num_samples=n_mcmc_samples,
+        num_chains=n_mcmc_chains,
+    )
+    mcmc.run(rng_key)
+
+
+@pytest.mark.parametrize(
+    "my_dist",
+    [
+        dist.TruncatedNormal(low=-1, high=1),
+        dist.TruncatedCauchy(low=-1, high=1),
+    ],
+)
+def test_repeated_predictive_method_in_truncated_distribution(my_dist):
+    n_data_samples = 10
+    n_prior_samples = 10
+    n_mcmc_samples = 10
+    n_mcmc_warmup = 10
+    n_mcmc_chains = 2
+    rng_key = jax.random.PRNGKey(0)
+
+    data_samples = my_dist.sample(rng_key, (n_data_samples,))
+
+    def my_model(obs=None):
+        numpyro.sample("obs", my_dist, obs=obs)
+
+    prior_predictive = Predictive(my_model, num_samples=n_prior_samples)
+    prior_predictive(rng_key)
```

Comment:
Is one call to `prior_predictive()` sufficient to reproduce the error without the patch, or are two calls needed, as the reproducible example from https://github.com/CDCgov/multisignal-epi-inference/issues/282?

Reply:
Thanks @dylanhmorris. The second call is initiated when running the inference. 

But your comments made me realize I don't need two separate tests; the second test can check if the chains run in parallel and two calls to predictive method (prior predictive and inference run here)

Reply:
closing this as resolved in https://github.com/pyro-ppl/numpyro/pull/1843/commits/82dbc935323bb352e1fc8c1d2e58cabaef523108